p2p/simulations/network.go

```diff
@@ -595,8 +585,21 @@ type Node struct {
 	// Config if the config used to created the node
 	Config *adapters.NodeConfig `json:"config"`
 
-	// Up tracks whether or not the node is running
-	Up bool `json:"up"`
+	// up tracks whether or not the node is running
+	up   bool
```

Comment:
we just need to make sure this field is marshalled with a json tag .
no custom marshaller needed (which added a lot of complexity)

Reply:
Fields which are not exported are not visible for the Marshaller. That was exactly the problem fixed by e7fdfbf923441b023c73cb39ac73a2c78f6e5c2e and the annotation was removed because it had no effect as [Janos pointed out](https://github.com/ethersphere/go-ethereum/pull/1178#discussion_r252668273)
Example: https://play.golang.org/p/pR6X1_pVbfT

Also note that the serialization is still kinda broken, as ~`Node.MarshalJSON().UnmarshalJSON()!=Node`. That is described in https://github.com/ethersphere/go-ethereum/issues/1177

Reply:
exactly because of the difference in unexported fields this equation is not expected to hold at all. 
Please avoid bringing in such code complexity for a simple thing that can be solved by renaming: Just use Up exported and if need lock, have `GetUp/SetUp`